dist/cargo_test.sh

```diff
@@ -23,16 +23,29 @@ executors["cargo"]="execute_native"
 executors["docker"]="execute_docker"
 
 function run_tests() {
+  HAS_KOV=0
+  if [[ $(type -f kcov && cargo kcov --version) ]]; then HAS_KOV=1; fi
   for directory in ${!cargo_test_flags[*]}; do
     (
-      set -x;
+      set -x
+      if [[ "${HAS_KOV}" ]]; then
+        # we want to prevent completely untested function to be stripped
+        export RUSTFLAGS='-C link-dead-code'
+        export CARGO_ARGS="kcov --verbose --all --no-clean-rebuild"
+      fi
+
       ${1} /usr/bin/env bash -c "\
         cd ${directory} && \
         export CARGO_TARGET_DIR=\"../target\" && \
-        cargo ${CARGO_ARGS:-test --release} ${cargo_test_flags[${directory}]} && \
+        cargo test --no-run &&
+        cargo ${CARGO_ARGS:-test} ${cargo_test_flags[${directory}]} && \
```

Comment:
I'm curious why you removed `--release` here. Would you mind to explain?

Reply:
This follows "hack/coverage.sh" script - don't release optimizations remove debug info so kcov won't be able to collect info? I'll check that in the debug commit